src/BundleAnalyzerPlugin.js

```diff
@@ -56,26 +68,33 @@ class BundleAnalyzerPlugin {
       } else if (this.opts.analyzerMode === 'static') {
         actions.push(() => this.generateStaticReport(stats));
       } else if (this.opts.analyzerMode === 'json') {
-        actions.push(() => this.generateStatsJSONReport(stats));
+        const budgetErrors = await this.generateStatsJSONReport(stats);
```

Comment:
shouldn't we put it to actions?

Reply:
ah, looks like it's async now, so you use await, ok

Reply:
I changed this hook to async hook, but we still need to report errors during the compilation phase so i had to remove JSON report from actions. I'm not sure if were needed since both this hooks were async hooks anyway.